buildSrc/subprojects/kotlin-dsl/src/main/kotlin/plugins/kotlin-dsl-module.gradle.kts

```diff
@@ -30,7 +30,8 @@ plugins {
 
 // including all sources
 tasks.jar {
-    from(sourceSets.main.get().allSource)
+    val main by sourceSets.getting
+    from(main.allSource - main.resources)
```

Comment:
Can you please add a comment about why this is necessary?

Reply:
Done